crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -4547,6 +4566,23 @@ impl<'db> BindingError<'db> {
             }
 
             Self::UnmatchedOverload => {}
+
+            Self::UnknownCallableSignature(callable_ty) => {
+                let node = Self::get_node(node, None);
+                if let Some(builder) = context.report_lint(&CALL_NON_CALLABLE, node) {
+                    let callable_ty_display = callable_ty.display(context.db());
+                    let mut diag = builder.into_diagnostic(format_args!(
+                        "Object of type `{callable_ty_display}` is not callable"
+                    ));
+                    diag.info(
+                        "An object with an unknown callable signature is not callable, \
+                        because there is no valid set of arguments for it",
```

Comment:
Oh, didn't realize until now that we have this -- that's great, but I think the wording here could also be improved to avoid saying its "not callable"

```suggestion
                        "This type includes all possible callables, so it cannot safely be called, \
                        because there is no valid set of arguments for it",
```